pkg/api/validate.go

```diff
@@ -24,7 +24,7 @@ var (
 		`/virtualNetworks/[^/]+` +
 		`$`)
 
-	rxAgentPoolProfileName = regexp.MustCompile(`(?i)^[a-z0-9]{1,12}$`)
```

Comment:
The docs seem to say they are case insensitive, why do this?

Reply:
Our converter merges on agent pool names and does not do so in a case insensitive way so we're actually broken here currently.  I am going to insert the agent pool name into the scale set name and don't want to litter the code everywhere with strings.EqualFold yet.  Follow-up in #939.